aten/src/TH/generic/THBlas.cpp

```diff
@@ -380,6 +383,39 @@ void THBlas_(gemm)(
     return;
   }
 #endif
+
+#ifdef USE_FBGEMM
+#if defined(TH_REAL_IS_LONG)
+  if (alpha == 1 && (beta == 0 || beta == 1)) {
+    bool accumulate;
+    if (beta == 0) {
+      accumulate = false;
+    } else {
+      accumulate = true;
+    }
+    // In FBGEMM, we assume row-major ordering; But here we use the column-major
+    // ordering following the FORTRAN tradition in BLAS interface. So we compute
+    // C^T = B^T * A^T instead.
+    fbgemm::cblas_gemm_i64_i64acc(
+        transb_ ? fbgemm::matrix_op_t::Transpose
+                : fbgemm::matrix_op_t::NoTranspose,
+        transa_ ? fbgemm::matrix_op_t::Transpose
+                : fbgemm::matrix_op_t::NoTranspose,
+        n,
+        m,
```

Comment:
in https://github.com/pytorch/FBGEMM/blob/f163b30ade066e2e5984ed3c8445ea18825d061b/src/FbgemmI64.cc
I see `M, N, K...` but here we have `n, m, k`. Are we correcting for some difference between naming conventions in  `Blas_(gemm)` vs fbgemm? Maybe that's what the comment above about fortran means but not very clear to me if that's the implication here.

Reply:
Right: this is related to the comments above. I have added more details in the comments.

Basically we can compute the GEMM with column major C (m x n) = A (m x k) * B (k x n) with the trick of transpose: C^T (n x m) = B^T (n x k) * A^T (k x m), which can be rewritten as the following MACRO: 
```
#define GEMM_COLMAJOR(transf_A, transf_B, m, n, k, alpha, A, lda, B, ldb, beta, C, ldc) \
        GEMM_ROWMAJOR(transf_B, transf_A, n, m, k, alpha, B, ldb, A, lda, beta, C, ldc)
```